### PR TITLE
[SID-1137] feat: add useOrganization and <OrganizationSwitcher />

### PR DESCRIPTION
**Pull request by jake-slashid**
## Description

[YouTrack issue](https://todo.irdcorp.dev/issue/SID-1137)

## Changes
- Add `useOrganizations()` hook
- Add `<OrganizationSwitcher />` component
- [Documentation](https://github.com/slashid/ng-evangelion/pull/930)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have generated the new version of the docs website and smoke tested it
- [x] I have checked that my changes haven't caused semantic errors in the existing docs
- [x] I have updated the README and DEVELOPMENT files

### CODE REVIEW

**Comment by jake-slashid** [comment]
For simplicity I'm considering
```tsx
<Dropdown
  disabled={isLoading}
  ...
/>
```

File: packages/react/package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
   },
   "devDependencies": {
     "@faker-js/faker": "^8.0.2",
-    "@slashid/slashid": "^3.9.0",
+    "@slashid/slashid": "link:/Users/jake/dev/ng-evangelion/web/sdk",
```
Comment (ikovic) [inline packages/react/package.json:58]: Leaving a note here so we don't forget replacing this with the actual package :)

File: packages/react/src/components/organization-switcher/index.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { useOrganizations } from "../../hooks/use-organizations"
+import { sprinkles } from "../../theme/sprinkles.css";
+import { Dropdown } from "../dropdown"
+import { Spinner } from "../spinner/spinner";
+
+export function OrganizationSwitcher() {
+  const { text } = useConfiguration();
+  const { organizations, currentOrganization, switchOrganization, isLoading } = useOrganizations()
+
+  if (isLoading || !currentOrganization) return <Spinner /> // need a better loading state
```
Comment (ikovic) [inline packages/react/src/components/organization-switcher/index.tsx:11]: This is the first time we offer a component that has to load something first - maybe it needs an optional loading fallback prop? Alternatively we could disable the input while it's loading 🤔

File: packages/react/src/context/slash-id-context.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ export type StorageOption = "memory" | "localStorage";
 
 export interface SlashIDProviderProps {
   oid: string;
+  onOidChange?: (oid: string) => void | Promise<void>
```
Comment (ikovic) [inline packages/react/src/context/slash-id-context.tsx:19]: Are we sure we need this? Maybe a direct callback could be registered on the hook itself?

File: packages/react/src/components/organization-switcher/index.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  return (
+    <Dropdown
+      defaultValue={currentOrganization.id}
+      className={sprinkles({ marginBottom: "3", marginTop: "5" })}
```
[the remark this answers is not in the document]
Comment (jake-slashid) [reply packages/react/src/components/organization-switcher/index.tsx:16]: Good idea 👍

File: packages/react/src/context/slash-id-context.tsx (hunk 3 above)
Comment (jake-slashid) [reply packages/react/src/context/slash-id-context.tsx:19]: It's not really needed to be honest, but equally neither is the reactive behaviour for the `oid` prop using `useEffect`.

I'm happy to treat `oid` as `initialOid` and be done with it, we can rename later.

File: packages/react/src/components/organization-switcher/index.tsx (hunk 2 above)
Comment (ikovic) [reply packages/react/src/components/organization-switcher/index.tsx:11]: Sounds great, thank you!

File: packages/react/src/hooks/use-organizations.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  const currentOrganization = useMemo(() => {
+    if (!user) return null
+
+    return organizations.find(org => org.id === user.oid) ?? null
+  }, [organizations, user, user?.oid])
+
+  useEffect(() => {
+    if (!user) return
+    if (organizations.length) return
```
Comment (kaspermroz) [inline packages/react/src/hooks/use-organizations.ts:24]: You could combine these I think

**Review by paulo-raca — COMMENTED** [review]
LGTM!

File: packages/react/src/hooks/use-organizations.ts (hunk 5 above)
Comment (jake-slashid) [reply packages/react/src/hooks/use-organizations.ts]: done, thanks! 🙌

File: packages/react/src/components/organization-switcher/index.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+const className = `sid-organization-switcher ${sprinkles({ marginBottom: "3", marginTop: "5" })}`
+
+const DefaultFallback = () => (
+  <Dropdown
+    defaultValue={undefined}
+    disabled={true}
+    className={className}
+    items={[]}
+    onChange={() => {}}
+    label={""}
```
[the remark this answers is not in the document]
Comment (jake-slashid) [reply packages/react/src/components/organization-switcher/index.tsx]: I added the label

**Comment by jake-slashid** [comment]
Documentation: https://github.com/slashid/ng-evangelion/pull/930

File: packages/react/src/components/organization-switcher/index.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        styles.organizationSwitcher,
+        "sid-organization-switcher"
+      )}
+    >
+      {children}
+    </div>
+  );
+};
+
+export const OrganizationSwitcher = ({
```
Comment (ikovic) [inline packages/react/src/components/organization-switcher/index.tsx:53]: Some JS doc would come in handy here 😃

**Review by ikovic — APPROVED** [review]
Just a minor comment about JS doc, otherwise looks great!

File: packages/react/src/components/organization-switcher/index.tsx (hunk 7 above)
Comment (jake-slashid) [reply packages/react/src/components/organization-switcher/index.tsx:65]: done

**Pull request merged**